### PR TITLE
Add catalog to web console

### PR DESCRIPTION
**Pull request by spadgett**
screen shot 2017-04-07 at 5 37 27 pm

@jwforres @jeff-phillips-18 FYI, not ready to merge, but it's working

### CODE REVIEW

**Comment by spadgett** [comment]
Requires https://github.com/openshift/origin-web-catalog/pull/73

File: app/scripts/services/navigate.js [hunk 1]
```diff
@@ -70,6 +70,10 @@ angular.module("openshiftConsole")
         return "project/" + encodeURIComponent(projectName) + "/overview";
       },
 
+      toProjectList: function(){
+        $location.path('projects');
```
Comment (jwforres) [inline app/scripts/services/navigate.js]: this should always work whether catalog is turned on or not, maybe we should add a redirect from projects to / when the landing page isn't enabled

Comment (spadgett) [inline app/scripts/services/navigate.js]: ok

File: app/scripts/app.js [hunk 2]
```diff
@@ -51,11 +52,23 @@ angular
...
     }
 
+    var landingPageRoute;
+    var projectsPageRoute = {
+      templateUrl: 'views/projects.html',
+      controller: 'ProjectsController'
+    };
+    if (_.get(window, 'OPENSHIFT_CONSTANTS.ENABLE_TECH_PREVIEW_FEATURE.service_catalog_landing_page')) {
+      landingPageRoute = {
+        templateUrl: 'views/landing-page.html',
```
Comment (jeff-phillips-18) [inline app/scripts/app.js]: Might want to use home.html rather than landing-page.html just for the look of the URL to users.

Comment (spadgett) [inline app/scripts/app.js]: The url should just be `/console`. Although maybe I shouldn't name the controller the same as the component

Comment (jeff-phillips-18) [inline app/scripts/app.js]: sorry, read it wrong.

Comment (jwforres) [inline app/scripts/app.js]: users should never see this URL, our templates are embedded into $templateCache at build time

**Comment by jwforres** [comment]
[merge]

**Pull request merged**